src/engine/cancun.md

```diff
@@ -120,11 +120,19 @@ Refer to the response for [`engine_forkchoiceUpdatedV2`](./shanghai.md#engine_fo
 
 #### Specification
 
-This method follows the same specification as [`engine_forkchoiceUpdatedV2`](./shanghai.md#engine_forkchoiceupdatedv2) with addition of the following:
+This method follows the same specification as [`engine_forkchoiceUpdatedV2`](./shanghai.md#engine_forkchoiceupdatedv2) with the following changes to the processing flow:
 
-1. Client software **MUST** check that provided set of parameters and their fields strictly matches the expected one and return `-32602: Invalid params` error if this check fails. Any field having `null` value **MUST** be considered as not provided.
+1. Client software **MUST** verify that `forkchoiceState` matches the [`ForkchoiceStateV1`](./paris.md#ForkchoiceStateV1) structure and return `-32602: Invalid params` on failure.
+
+2. Extend point (7) of the `engine_forkchoiceUpdatedV1` [specification](./paris.md#specification-1) by defining the following sequence of checks that **MUST** be run over `payloadAttributes`:
+
+    1. `payloadAttributes` matches the [`PayloadAttributesV3`](#payloadattributesv3) structure, return `-38003: Invalid payload attributes` on failure.
```

Comment:
It's not clear how we extend point 7 of paris fork. Might be better just to add this rules to paris also in expected order

Reply:
One of this rules is in Paris which is checking the `timestamp` wrt parent block’s timestamp. Other two are introduced in Cancun, so they aren’t relevant for Paris. The point (7) in Cancun has an extended list of checks wrt to Paris

Reply:
I also had some confusion wrt to this. We have an engine api test:
```
ForkchoiceUpdatedV2 To Request Shanghai Payload, Non-Null Beacon Root
```
where we essentially fcuV2 before Cancun, with a Shanghai `timestamp` and a non null `parentBeaconBlockRoot` (incorrect Payload Attributes structure).

As we are before Cancun my assumption would be that the extended points to (7) are not yet applied. Most clients return `-38003: Invalid payload attributes` for this case. The addition to Paris only validates the Payload Attributes based on the timestamp. So to my understanding we don't explicitly state what to return in this case.

So I agree with @flcl42 that we could add the Payload Attributes structure check to Paris. Apologies if I am missing something here

Reply:
why should the `parent.timestamp` check be removed from cancun?
Also, i am hesitant to change the previous versions of any methods in the spec unless the change is coherent with the behavior of the majority of EL clients

Reply:
Do EL clients check `payloadAttributes` structure in case of V1? If we remove a check from V3 then it is not clear in which order they are applied which may result in the return error message to be different on the same payloadAttributes in different client implementations

Reply:
> why should the parent.timestamp check be removed from cancun?
My fault - it should be kept, I wasn't taking into consideration the order of the checks!

I'd like to double check our tests for the `payloadAttributes` structure in case of V1. Will get back to you :)